src/advancedKubectl/content.md

```diff
@@ -0,0 +1,1108 @@
+
+The job of the ReplicaSet controller is to make sure that the required number of replica Pods of a ReplicaSet exists. In our example, no Pods exist yet, so the ReplicaSet controller creates these Pod definitions (according to the Pod template in the ReplicaSet definition) and saves them in the storage backend.
+
+The creation of the new Pods triggers the **scheduler**, who watches for Pod definitions that are not yet scheduled to a worker node. The scheduler chooses a suitable worker node for each Pod and updates the Pod definitions in the storage backend with this information.
+
+> Note that up to this point, no workload code is being run anywhere in the cluster. All that has been done so far is creating and updating resources in the storage backend on the master node.
+
+This event triggers the **kubelets** who watch for Pods that are scheduled to their worker nodes. The kubelet of the worker node your ReplicaSet Pods have been scheduled to downloads the container images for these Pods (if not already present on the machine) and runs them with the configured container runtime (which may be Docker or another container runtime).
+
+At this point, finally, your ReplicaSet application is running!
+
+### The role of the Kubernetes API
+
+As you can see from the above example, Kubernetes components (except the API server and the storage backend) work by watching for resource changes in the storage backend and manipulating resources in the storage backend.
+
+However, these components **do not** access the storage backend directly, but only **through the Kubernetes API**.
+
+Consider the following examples:
+
+- The ReplicaSet controller uses the [*list ReplicaSets* API endpoint](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#list-replicaset-v1-apps) API operation with a `watch` parameter for watching for changes to ReplicaSet resources.
+- The ReplicaSet controller uses the [*create Pod* API endpoint](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#create-pod-v1-core) for creating Pods.
+- The scheduler uses the [*patch Pod* API endpoint](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#patch-pod-v1-core) for updating Pods with the information about the selected worker node.
+
+As you can see, this is the **same API that is also used by kubectl**.
+
+This double usage of the Kubernetes API for internal components as well as for external users is a **fundamental design concept** of Kubernetes.
+
+With this knowledge, you can summarise how Kubernetes works as follows:
+
+- The storage backend stores the state (i.e. resources) of Kubernetes.
+- The API server provides an interface to the storage backend in the form of the Kubernetes API.
+- All other Kubernetes components and users read, watch, and manipulate the state (i.e. resources) of Kubernetes through the Kubernetes API.
```

Comment:
feel free to ignore, but perhaps you could have this as an ordered list.

Reply:
Could make sense.